pkg/kubelet/kuberuntime/kuberuntime_manager.go

```diff
@@ -845,6 +842,61 @@ func (m *kubeGenericRuntimeManager) doPodResizeAction(pod *v1.Pod, podContainerC
 	return resizeResult
 }
 
+// validatePodResizeAction checks whether the proposed resize actions are currently viable.
+func (m *kubeGenericRuntimeManager) validatePodResizeAction(
+	ctx context.Context,
+	pod *v1.Pod,
+	podStatus *kubecontainer.PodStatus,
+	currentPodResources, desiredPodResources *cm.ResourceConfig,
+	podContainerChanges podActions,
+) error {
+	if len(podContainerChanges.ContainersToUpdate[v1.ResourceMemory]) > 0 || podContainerChanges.UpdatePodResources {
+		podLimitDecreasing := desiredPodResources.Memory != nil && (currentPodResources.Memory == nil || // Pod memory limit added
+			*desiredPodResources.Memory < *currentPodResources.Memory) // Pod memory limit decreasing
+		decreasingContainerLimits := map[string]int64{} // Map of container name to desired memory limit.
+		for _, cUpdate := range podContainerChanges.ContainersToUpdate[v1.ResourceMemory] {
+			if cUpdate.desiredContainerResources.memoryLimit != 0 {
+				if cUpdate.currentContainerResources == nil || cUpdate.currentContainerResources.memoryLimit == 0 {
+					// Container memory limit added
+					decreasingContainerLimits[cUpdate.container.Name] = cUpdate.desiredContainerResources.memoryLimit
+				} else if cUpdate.desiredContainerResources.memoryLimit < cUpdate.currentContainerResources.memoryLimit {
+					// Container memory limit decreasing
+					decreasingContainerLimits[cUpdate.container.Name] = cUpdate.desiredContainerResources.memoryLimit
+				}
+			}
+		}
+		if podLimitDecreasing || len(decreasingContainerLimits) > 0 {
+			podUsageStats, err := m.runtimeHelper.PodCPUAndMemoryStats(ctx, pod, podStatus)
+			if err != nil {
+				return fmt.Errorf("unable to read memory usage for pod %s", format.Pod(pod))
+			}
+			var errs []error
+			if podLimitDecreasing {
+				if podUsageStats.Memory == nil || podUsageStats.Memory.UsageBytes == nil {
+					errs = append(errs, fmt.Errorf("missing pod memory usage"))
+				} else if *podUsageStats.Memory.UsageBytes >= uint64(*desiredPodResources.Memory) {
+					errs = append(errs, fmt.Errorf("attempting to set pod memory limit below current usage (%d)", *podUsageStats.Memory.UsageBytes))
```

Comment:
nit: would it be helpful to include the desired pod memory limit in this error message?